packages/angular_devkit/schematics/src/rules/utils/is-binary.ts

```diff
@@ -5,24 +5,16 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+import { TextDecoder } from 'util';
 
-export function isBinary(buffer: Buffer): boolean {
-  const chunkLength = 24;
-  const chunkBegin = 0;
-
-  const chunkEnd = Math.min(buffer.length, chunkBegin + chunkLength);
-  const contentChunkUTF8 = buffer.toString('utf-8', chunkBegin, chunkEnd);
+const decoder = new TextDecoder('utf-8', { fatal: true });
 
-  // Detect encoding
-  for ( let i = 0; i < contentChunkUTF8.length; ++i ) {
-    const charCode = contentChunkUTF8.charCodeAt(i);
-    if ( charCode === 65533 || charCode <= 8 ) {
-      // 8 and below are control characters (e.g. backspace, null, eof, etc.).
-      // 65533 is the unknown character.
-      return true;
-    }
+export function isBinary(buffer: Buffer): boolean {
+  try {
+    // `TextDecoder.prototype.decode` will return a UTF-8 string
+    // if there are no decoding errors occured.
+    return !decoder.decode(buffer);
```

Comment:
Attempting to decode the entire file is going to be highly inefficient.  The  buffer size could be unbounded (megabytes or even gigabytes).  A minimal amount of bytes should be analyzed when possible.  If 24 bytes is insufficient this could be increased probably up to even a kilobyte if needed.

Reply:
Agreed. I was actually thinking about some external package that provides such type of functionality. I don't know if it's a good idea to bring some package to this codebase.

The problem is that if we select a chunk based on the number of bytes then yes - it will not work correctly. But it will work if we select it based on number of characters.

https://github.com/parshap/truncate-utf8-bytes/blob/v1.0.2/lib/truncate.js

Seems like it's being used a lot regarding "used by" button. It doesn't work with buffers but we could copy-paste and change it a little bit, it's short.

Just thinking aloud...

Reply:
Based on the singular usage of this function, I think we should consider removing it completely and instead adjust the logic within the function that uses this helper (https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/schematics/src/rules/template.ts#L52).  Direct usage of the `TextDecoder` to get the content combined with a try/catch block that handles the invalid data error should avoid the entire issue.

Reply:
Removed `is-binary.ts` file and moved all logic in the `template.ts` file.